lib/PerceptionManager.tsx

```diff
@@ -59,47 +64,82 @@ export const perceptionManager: IPerceptionManager = (() => {
         });
     }
 
+    const buildSceneTurtleData = (things: Thing[]): Promise<string> => {
+        const scene: SolidDataset = createSolidDataset();
+
+        for(let thing of things) {
+            setThing(scene, thing);
+        }
+
+        return triplesToTurtle(Array.from(scene));
+    }
+
+    /**
+     * builds a list of ITerminalMessage objects by getting the primary content data from parameterised things
+     */
+    const getPrimaryContent = (things: Thing[]) : ITerminalMessage[] => {
+        let messages: ITerminalMessage[] = [];
+        
+        for(let thing of things) {
+            const uri = asUrl(thing as ThingPersisted);
+            const imageUrl = getUrl(thing, MUD.primaryImageContent);
+
+            if(imageUrl && !recentUris.includes(uri)) {
```

Comment:
This conditional - do we actually want msg below to be added if this is falsy?

Reply:
I think so most of the time, e.g. if I visited the bar a minute ago and then I go back in, I still want it to be described

It'd be nice to have it as a parameter to this function I suppose, and not even "on/off" but maybe something like `ContentLevelEnum` ... but I think that's a future development